conans/client/tools/net.py

```diff
@@ -9,10 +9,13 @@
 _global_requester = None
 
 
-def get(url, md5='', sha1='', sha256='', destination="."):
+def get(url, md5='', sha1='', sha256='', destination=".", filename=""):
     """ high level downloader + unzipper + (optional hash checker) + delete temporary zip
     """
-    filename = os.path.basename(url)
+    if not filename and any(char in url for char in ["?", "="]):
```

Comment:
``if not filename and ("?" in url or "=" in url)``
IMHO simpler to read, avoids call to any + list creation + iteration